app/jobs/calendar_importer/calendar_importer.rb

```diff
@@ -16,19 +16,34 @@ class InaccessibleFeed < StandardError; end
     CalendarImporter::Parsers::Ticketsolve
   ].freeze
 
+  MANUAL_PARSERS = (DETECTABLE_PARSERS.dup.concat [
+    CalendarImporter::Parsers::LdJson
+    # CalendarImporter::Parsers::RssFeed
+  ]).freeze
+
   def initialize(calendar)
     @calendar = calendar
 
     validate_feed!
   end
 
   def parser
-    @parser ||=
-      if @calendar.importer_mode == 'auto'
-        PARSERS.find { |parser| parser.handles_url?(@calendar.source) }
-      else
-        PARSERS.find { |parser| parser::KEY == @calendar.importer_mode }
+    return @parser if @parser
+
+    if @calendar.importer_mode == 'auto'
+      @parser = DETECTABLE_PARSERS.find { |parser| parser.handles_url?(@calendar.source) }
+
+      if @parser.blank?
+        # TODO: now try to detect ld+json
+        try_parser = CalendarImporter::Parsers::LdJson.new(@calendar)
+        nodes = try_parser.download_calendar
+        @parser = CalendarImporter::Parsers::LdJson if nodes.present?
       end
+
+      return @parser
+    end
+
+    @parser = MANUAL_PARSERS.find { |parser| parser::KEY == @calendar.importer_mode }
```

Comment:
I'm not sure we need all this? Can we just make `CalendarImporter::Parsers::LdJson.handles_url?` do:
```rb
try_parser = CalendarImporter::Parsers::LdJson.new(@calendar)
try_parser.download_calendar.present?
```
or similar and leave the rest of the detection logic as it was?

Reply:
Doing this now